stdlib/public/core/Character.swift

```diff
@@ -399,8 +399,7 @@ extension StringProtocol {
           UTF8.self, input: smallUTF8))
       
     case let .large(value):
-      let buf = String(_StringCore(_StringBuffer(value)))
-      self = Self(buf[buf.startIndex..<buf.index(after: buf.startIndex)])
+      self = Self(String(_StringCore(_StringBuffer(value))))
```

Comment:
Do you think StringProtocol should have an (internal) init directly from a _StringCore?